clang/test/CIR/CodeGenBuiltins/X86/avx512bw-builtins.c

```diff
@@ -768,3 +756,51 @@ unsigned char test_ktestz_mask64_u8(__mmask64 A, __mmask64 B) {
   return _ktestz_mask64_u8(A, B);
 }
 
+
+
+__m512i test_mm512_movm_epi16(__mmask32 __A) {
+  // CIR-LABEL: _mm512_movm_epi16
+  // CIR: %{{.*}} = cir.cast bitcast %{{.*}} : !u32i -> !cir.vector<32 x !cir.int<u, 1>>
+  // CIR: %{{.*}} = cir.cast integral %{{.*}} : !cir.vector<32 x !cir.int<u, 1>> -> !cir.vector<32 x !s16i>
+  
+  // LLVM-LABEL: @test_mm512_movm_epi16
+  // LLVM:  %{{.*}} = bitcast i32 %{{.*}} to <32 x i1>
+  // LLVM:  %{{.*}} = sext <32 x i1> %{{.*}} to <32 x i16>
```

Comment:
I think this test, among others, will fail because of the sign-extend issue. I'll wait until I've merged that before starting the test run for this PR. Hopefully you won't have to make any changes.

Reply:
Oh, I was wrong. It looks like you'll need to rebase and update the checks in this PR that are looking for `<N x !cir.int<u, 1>>`